### DISCUSSION TITLE
How to change the request data in form

### DISCUSSION THREAD

**Discussion by Guervyl** [symfony/symfony#49252] [Q&A] [answered]
I want to change the form submitted data in [The FormEvents::PRE_SUBMIT Event](https://symfony.com/doc/current/form/events.html#a-the-formevents-pre-submit-event) as the document says: 

**It can be used to:
Change data from the request, before submitting the data to the form;**

But using the `RequestStack->getCurrentRequest()->replace() | add() | set()` does not change the form data.

**Comment by MatTheCat** [comment, marked as answer]
Your listener will receive an instance of `FormEvent`. You can get the form data set from the request with its `getData` method, and change them using `setData`.

See the [`TrimListener`](https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Form/Extension/Core/EventListener/TrimListener.php) for a simple example.

**Reply by Guervyl** [reply]
I think that should be in the documentation. Because I was confused about that I should get the Request and change the data. I spent hours on this. And it was so simple.
